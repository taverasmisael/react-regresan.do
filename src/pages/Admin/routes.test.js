import React from 'react'
import { MemoryRouter } from 'react-router-dom'

import { shallow, mount } from 'enzyme'

import Routes from './routes'
import { OverviewPage, BranchesPage } from './'

describe('Pages/Admin/routes', () => {
  describe('rendering', () => {
    it('renders without crashing', () => {
      shallow(<Routes />)
    })
    it('matches snapshot', () => {
      expect(shallow(<Routes />)).toMatchSnapshot()
    })
  })

  describe('functionality', () => {
    it('redirects to /admin if not matching route is found', () => {
      const component = mount(
        <MemoryRouter initialEntries={['/fakepath']}>
          <Routes />
        </MemoryRouter>
      )
      expect(component.find(OverviewPage)).toHaveLength(1)
    })
    it('renders "Overview"', () => {
      const component = mount(
        <MemoryRouter initialEntries={['/admin/']}>
          <Routes />
        </MemoryRouter>
      )
      expect(component.find(OverviewPage)).toHaveLength(1)
    })
    it('renders "Branches"', () => {
      const component = mount(
        <MemoryRouter initialEntries={['/admin/branches']}>
          <Routes />
        </MemoryRouter>
      )
      expect(component.find(BranchesPage)).toHaveLength(1)
    })
  })
})
