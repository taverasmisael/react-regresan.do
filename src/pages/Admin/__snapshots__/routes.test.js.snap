// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Admin/routes rendering matches snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/admin"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/admin/branches"
  />
  <Redirect
    push={false}
    to="/admin"
  />
</Switch>
`;
